Redesigned the web debug toolbar

javiereguiluz opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

In my opinion, the design of the current Web Debug Toolbar suffers from two problems: it looks outdated and it lacks polishing on some parts. Symfony has always introduced a big update in this toolbar for each big release, so I propose to do the same for Symfony 3.

Let's see the new proposed toolbar in action:

---

**This is the default toolbar** (click on the image to enlarge it)

![default](https://cloud.githubusercontent.com/assets/73419/8449444/cb4be4ea-1fd1-11e5-9358-332449375531.png)

As you can see, if some panel doesn't provide any information, we don't show it (e.g. 0 queries, 0 forms, 0 logs, 0 ajax requests).

---

Let's compare some panels in detail:

**Symfony Panel**

![symfony-old-panel](https://cloud.githubusercontent.com/assets/73419/8449504/407302b2-1fd2-11e5-80da-a49dc99e68f5.png) ![symfony-new-panel](https://cloud.githubusercontent.com/assets/73419/8449505/41698e20-1fd2-11e5-8ced-e2717716c2d3.png)

**Request Panel**

![request-old-panel](https://cloud.githubusercontent.com/assets/73419/8449531/7d6888ea-1fd2-11e5-829e-93c9e50b7b64.png) ![request-new-panel](https://cloud.githubusercontent.com/assets/73419/8449532/7e359da8-1fd2-11e5-87e6-f194d824254c.png)

---

When needed, more panels are displayed, such as the SQL information:  (click on the image to enlarge it)

![sql_panel](https://cloud.githubusercontent.com/assets/73419/8449554/b6abc1d0-1fd2-11e5-8450-13eae5bc3c50.png)

As you can see, the new toolbar provides more information than the old one and it takes less space.

Some of the new panels include more information in the extended info, such as the Doctrine one showing that the second level cache is disabled:

![sql_old_panel](https://cloud.githubusercontent.com/assets/73419/8449573/fc288aae-1fd2-11e5-9823-46b33fa9d998.png) ![sql_new_panel](https://cloud.githubusercontent.com/assets/73419/8449572/fab6244c-1fd2-11e5-9e14-28045de0b143.png)

---

**Errors and warnings** now stand out more clearly because all the panel background is changed.

For example, if there are i18n errors: (click on the image to enlarge it)

![i18n_panel_error](https://cloud.githubusercontent.com/assets/73419/8449590/20708786-1fd3-11e5-8ad7-6bd3b3f4b933.png)

If the page is loading too slowly: (click on the image to enlarge it)

![slow_page](https://cloud.githubusercontent.com/assets/73419/8449597/2bc0c6e6-1fd3-11e5-9a9e-55bfe07ac170.png)

If Symfony version is deprecated: (click on the image to enlarge it)

![deprecated_symfony](https://cloud.githubusercontent.com/assets/73419/8449605/3cfa9860-1fd3-11e5-8dbd-9915597970f4.png)

---

HTTP Errors also stand out more clearly: (click on the image to enlarge it)

![error_404](https://cloud.githubusercontent.com/assets/73419/8449617/56a4ccb8-1fd3-11e5-9638-322f1840937c.png)

![error_500](https://cloud.githubusercontent.com/assets/73419/8449618/57a79294-1fd3-11e5-8383-2598dee73fc6.png)

---

Some questions that you may be wondering:
- **Why use a dark toolbar instead of maintaining the light toolbar?** Because a dark toolbar stands out more from most of the web designs. It's more probable that your applications display a light background than a dark background, so the Symfony Toolbar stands out more if it's designed in dark.
- **What about the profiler?** If this proposal is approved I'll also update the design of the profiler to match this new dark and modern look-and-feel.
- **What about smaller screens?** This is a proposal, so I haven't finished it. Tweaking the design for smaller screens will be the next step. Anyway, as you can see the new toolbar already takes much less space than the current one, so it won't be hard to adapt it.

javiereguiluz committed:
Redesigned the web debug toolbar

aderuwe commented:
Very nice, I like it!

stof commented:
How would it look on smaller screens ? currently, we keep only the icons, but you dropped them. So I think it may still be hard :smile: 

However, I like the idea. I have 1 request though: please make it easy for third-party bundles to provide toolbar integrations compatible with both 2.x and 3.x, otherwise most bundles will provide a crappy integration on some versions (these bundles will need to keep supporting the 2.7 and 3.8 LTS for a while).
and btw, should we include it in 3.0 only ? I would vote to have it in 2.8 instead if we can make it BC for bundles providing panels.

kashike commented:
This looks wonderful, good job!

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig:
```diff
@@ -1,28 +1,35 @@
+    {% elseif 'dev' == collector.symfonyState %}
+        {% set block_status = 'yellow' %}
+        {% set symfony_version_status = 'This Symfony version is still in the development phase.' %}
+    <span class="sf-toolbar-status sf-toolbar-status-yellow" title="">
```
is it expected to have this opening span here ? I cannot see where you close it

WouterJ replied:
this should indeed be removed. It's now also causing the close button to not work properly anymore.

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig:
```diff
@@ -1,28 +1,35 @@
+            {% elseif collector.symfonyState is defined %}
+                <span class="sf-toolbar-label">symfony</span>
+                <span class="sf-toolbar-value">{{ collector.symfonyversion }}</span>
+            {% endif %}
```
the link may end up empty here, which looks weird

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -433,11 +472,3 @@
     }
 }
-
-/***** Media query print: Do not print the Toolbar. *****/
```
this should be kept IMO

javiereguiluz replied:
I can't imagine anyone printing a web page in the `dev` environment. But if someone ever did, don't you think that they might actually want to print the entire page, including the toolbar?

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar_item.html.twig:
```diff
@@ -1,9 +1,12 @@
+
+    <div class="sf-toolbar-block sf-toolbar-block-{{ block_name|default(name) }} sf-toolbar-status-{{ block_status|default('') }}">
```
the indentation seems off

jrobeson commented:
One of the problems with the current web debug toolbar is that it is almost actively hostile to Content Security Policy. I'm not suggesting that you put in any effort to work with it, but It'd be nice if you could make sure to keep any twig related logic out of the javascript code, so it's at least easier to override.

stof commented:
There is another weird thing: you have removed some icons, but not all of them.

> and btw, should we include it in 3.0 only ? I would vote to have it in 2.8 instead if we can make it BC for bundles providing panels.

looks like you actually agree with me @javiereguiluz as you submitted it to 2.8 :smile: 

@jrobeson please open a separate issue about the CSP compatibility. It is not related to a redesign.

jrobeson commented:
yeah i was just about to retract my comment. This PR didn't make the situation worse.

javiereguiluz commented:
@stof if this proposal is approved, I'm going to reintroduce all icons to display it in smaller screens.

xavierleune commented:
Hi,

This is a very good idea. The new look & feel is prettier but I find it harder to read. If icons are back, like you suggest it in your previous comment, that will be just fine.

So that's a :+1:  for symfony 3.0 for me.

Pierstoval commented:
:+1: for any Symfony version, it looks awesome and reminds me the look of all SensioLabs related websites :+1:

aitboudad commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig:
```diff
@@ -57,14 +54,6 @@
 {% endblock %}
 
 {% block panelContent %}
-    {% set filter = request.query.get('state', '-1') %}
```
This code is for adding the translation profiler (see # 14993 ), is there a reason to remove it or just by mistake ?

javiereguiluz replied:
It's a mistake. I'm sorry. If this proposal is approved I'll review each panel very carefully. Thanks.

WouterJ commented:
A thousand times :+1:. Improving it's UI has been on my create-an-issue-about-it list for a long time :)

I like the black background, this seems to be the industry standard for toolbars (see WordPress, Drupal, StackOverflow, etc.).

However, white text on a black background is harder to read than black text on a white background. I think we can improve UI of the toolbar by making the text some pixels bigger (at least the white text). Some text (like "anon.") a hard to read.
Also, the close icon seems to be very large, compared to other icons/text. I think it should be smaller than or the same as the rest of the icons.

Green-Cat commented:
:+1:

Baachi commented:
Big :+1: for this. Love it :heart:

dosten commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -180,7 +180,7 @@
                 requestCounter[0].textContent = requestStack.length;
 
-                var className = 'sf-toolbar-ajax-requests sf-toolbar-status';
+                var className = 'sf-toolbar-ajax-requests sf-toolvar-value';
```
typo? toolVar

WouterJ commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -26,39 +20,32 @@
-.sf-toolbarreset th {
-    font-size: 11px;
+    border: none;
+    cursor: text;
```
please use `cursor: pointer;` for abbreviations as well. Currently, the toolbar is a pointer everywhere except from the abbreviations. As the abbreviations are still clickable, this mouse change is quite strange.

stof replied:
well, or at least don't reset the cursor at all, letting it become a pointer because of the abbr being inside a link. forcing ``cursor: pointer`` on all abbr is not good either in case they are not clickable

stof replied:
@javiereguiluz ping on this comment

javiereguiluz replied:
I've restored the previous design to show a border-bottom for abbr:

![abbr_dashed_border](https://cloud.githubusercontent.com/assets/73419/9008031/8646b932-3797-11e5-8b23-10b2666a7c90.png)

WouterJ commented:
Btw, nothing in the bar now indicates that's a toolbar from Symfony. I would prefer if the symfony logo was added somewhere in the bar (right/left), just like in the old toolbar.

mickaelandrieu commented:
:+1:  and I (also) agree this can be part of  Symfony 2.8

iltar commented:
You have my vote, this looks a lot more user-friendly to me! :+1:

romqin commented:
I'm :+1: with it even if I'm not really fan of the new redesign.

There is a small bug:
![capture d ecran 2015-07-02 09 49 21](https://cloud.githubusercontent.com/assets/334432/8472242/cf5ca582-209f-11e5-9bea-e155c947f780.png)

When I want to hide the web debug toolbar:
![capture d ecran 2015-07-02 09 49 29](https://cloud.githubusercontent.com/assets/334432/8472243/cf683f00-209f-11e5-8e12-7d821f71283c.png)

javiereguiluz commented:
Thank you all for your comments! So far the feedback from the communty has been mostly positive. Next steps:

1) Let's wait a few hours/days for the Symfony Deciders to vote on this proposal.
2) If approved, I'll finish it, fix all the reported issues, etc.

aitboudad commented:
I'm :+1:

iltar commented:
@javiereguiluz maybe it's more about the details, but the "invalid entities" green rounded corner square is not in style with the rest anymore. Maybe it's an idea to just make the "0" green or the "N" red instead?

fabpot commented:
I'm also interested in seeing what we can come up with by redesigning the toolbar.

xabbuh commented:
Looks good so far.

mickaelandrieu commented:
@javiereguiluz also, there is a WIP to make the Profiler a standalone component: does this WIP have an impact on your redesign ?

I mean, maybe we can have a new component in 2.8, which introduce a new design.
Also, is there a way to make a nice redesign *without* remove any html but by adding a new css file and add some new informations ?

I'm not sure we need to break the actual HTML in order to make a new UI, what do you think ?

mickaelandrieu commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -165,43 +190,67 @@
+    padding-left: 0;
 }
 
+/*
```
don't forget to remove this CSS block.

iltar commented:
> Last question: in the database panel, when we read "3sql/XXms" is it to add the information on the actual database engine used ? For example, what is the behavior if I use mongoDB ?

It might be a little bit too generic, I have some applications that have both propel and doctrine (legacy + migration), this would yield "incorrect" results.

stof commented:
I'm also in favor of continuing this redesign.

@javiereguiluz please try to keep BC on the HTML markup needed for bundles hooking in the profiler, with a way to support both the 2.3 and the 2.8 designs

javiereguiluz commented:
@stof yes, let's try to keep BC. I plan to update all the official bundles, so I guess you refer to third-party bundles. My question: would be enough to make sure that there are some CSS rules that make the old HTML panels look good enough in the new toolbar?

stof commented:
@javiereguiluz even "official" bundles. I want to keep support for Symfony 2.7 in DoctrineBundle without having multiple branches. this means that there should be a way to have a markup compatible with both CSS without hassle (it is good for me if bundles are encouraged to do some changes to make their panel look better in the new design, for instance by adding a few more classes in some places, as long as the markup still looks good in the old design). So the idea is to have the same markup being good enough in both design (and more than good enough in one of them)

mickaelandrieu commented:
@stof @javiereguiluz yes ! :+1:

javiereguiluz committed:
First batch of fixes

guilhermeblanco commented:
@javiereguiluz I haven't tried the patch yet, but it would be nice if we handled nicely a couple of odd cases:

* When many panels are used (Doctrine, Solr, Security, Translation and so on), currently it overflows to another line. The only current fix is to find a window size that compresses the information, allowing you to not cause line break on toolbar.
My suggestion here would be an approach similar to this ( http://jsfiddle.net/jbrosi/c6kf2/ ) which calculates the size of the next visible item and scroll it entirely. That would prevent the double sized debug toolbar because every bundle has its own panel.

* When resizing the screen, it would be nice to shrink the information visible at a glance, the same way we do currently. Currently, as you reduce the screen size, we show less and less information until we reach the point where only icons are being displayed.

Also I'd share some personal suggestions:

* There's no need to display PHP and Symfony versions. A mouse over already provides this information and this is not an every day task to look at them. Folding this information should give an extra breathe of space for more tooling for daily tasks.

* And use their icons instead of text. =)

* Keep the icons for every panel, as shrinking the window size should at the end keep only the icons

* Use route names instead of Controller:Action. They exist to simplify our lives, generally they're shorter than the other and hovering it should give all info we need. I'd say we could even remove it (and rely on hover of status code as first resizing strategy)

Pierstoval commented:
> Use route names instead of Controller:Action. They exist to simplify our lives, generally they're shorter than the other and hovering it should give all info we need. I'd say we could even remove it (and rely on hover of status code as first resizing strategy)
 
I'd recommend using both like now, because in very complex apps you need the controller and action in order to remember the bundle (and then the "concept") in which is handled the action, especially when you are using the shortcuts with `framework: { ide: "..." }` config options, which is already implemented in the debug toolbar

stof commented:
@Pierstoval but do they need to be in the toolbar itself, or in the extra information added on hover ? currently, the controller and route are taking 30% of the width of my toolbar, while all other infos are small

romqin commented:
> I'd recommend using both like now
In fact, we removed the route name. See #14903

stof commented:
ah indeed. I missed that the route name is now only displayed in the hover panel.

javiereguiluz committed:
Restored all the code removed by mistake

ogizanagi commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig:
```diff
@@ -16,26 +16,30 @@
             </div>
             <div class="sf-toolbar-info-piece">
+                <b>Controller file</b>
+                <span>{{ collector.controller.class }}</span>
```
When controller does not exists: 
> Impossible to access an attribute ("class") on a string variable ("n/a") in @WebProfiler/Collector/request.html.twig at line 38

javiereguiluz committed:
A new batch of updates

javiereguiluz committed:
Fixed a minor markup error that broke the toolbar

javiereguiluz committed:
Reorder toolbar panels

javiereguiluz committed:
More fixes and tweaks

javiereguiluz commented:
Here it is a quick update of the latest changes made to this PR:

**1)** The config panel has been merged into a single panel (you'll no longer see the PHP panel). And the Symfony logo is back for branding reasons and to save space:

![config_panel](https://cloud.githubusercontent.com/assets/73419/8568116/35771ad0-256b-11e5-9576-3d352b7df964.png)

All the previous PHP information is now included in the "info panel":

![config_panel_reveal](https://cloud.githubusercontent.com/assets/73419/8568126/54ff3bda-256b-11e5-8010-a420015f0395.png)

**2)** The request panel no longer displays the controller + method, but just the route name:

![request_panel](https://cloud.githubusercontent.com/assets/73419/8568133/6578920e-256b-11e5-858e-52a46d76c70b.png)

I know that this may be shocking for some of you, but this is "mandatory" to save space and therefore display all the toolbar's panels. Again, the rest of the information is available in the "info panel":

![request_panel_review](https://cloud.githubusercontent.com/assets/73419/8568155/94e69400-256b-11e5-8bb5-db42692cb15c.png)

**3)** Minor cleaning changes (we use `include()` instead of `{% include %}`, the inlined SVG icons have been moved to individual templates, we have updated the CSS code (for example to use `display: table` instead of some ugly hacks), etc.)

xabbuh commented:
> 2) The request panel no longer displays the controller + method, but just the route name: [...]

Is it really necessary to save so much space? I would be more happy if the toolbar displayed the controller class and method name instead of the route name (most of the time I am more interested in the concrete method that is responsible for serving a request).

javiereguiluz commented:
I don't know if this is the best solution to our problem ... but it's true that we have a space problem that needs to be solved.

Take a look at the typical toolbar of a common page:

![crowded_panel](https://cloud.githubusercontent.com/assets/73419/8568622/a14f5890-256f-11e5-8181-51385916ceb2.png)

Even if the screen resolution is generous, we are a bit short of space, because that toolbar doesn't include yet the panel for Ajax requests, Swiftmailer, exception/logs and all the third-part panels (Sonata, etc.)

javiereguiluz committed:
Lots of minor improvements

javiereguiluz committed:
Finished the Ajax panel redesign

javiereguiluz committed:
Minor Ajax tweaks

javiereguiluz committed:
Improved the Security toolbar panel

javiereguiluz committed:
Tweaked the translation panel

javiereguiluz committed:
Tweaked the Twig panel

javiereguiluz committed:
Minor CSS fixes

javiereguiluz committed:
Finished "dump" panel and other minor tweaks

javiereguiluz committed:
Finished the toolbar redesign

javiereguiluz committed:
Fixed a minor syntax issue

javiereguiluz commented:
**The toolbar redesign has just been completed**, so you can fully review it and test it. Please report bugs and missing features. Thanks!  Note: I'm aware of some CSS vertical aligning issues.

A quick overview of the final version:

### When the toolbar is minimized

![toolbar-minimized](https://cloud.githubusercontent.com/assets/73419/8600418/2b232096-2666-11e5-9d3c-26029d1cc4bb.png)

### Simplest toolbar possible

Panels without information are not shown:

![toolbar-minimal](https://cloud.githubusercontent.com/assets/73419/8600434/35aadfd6-2666-11e5-8767-600e7cef3a8e.png)

### Responsive design

#### Smartphones

![toolbar-mobiles](https://cloud.githubusercontent.com/assets/73419/8600477/6d5f222a-2666-11e5-84d0-01d941691399.png)

#### Tablets

![toolbar-tablets](https://cloud.githubusercontent.com/assets/73419/8600495/83bfa3a0-2666-11e5-82e8-1d35786ab04e.png)

#### 1024px

![toolbar-1024](https://cloud.githubusercontent.com/assets/73419/8600482/760c6720-2666-11e5-86e2-125f71a2d461.png)

#### 1200px

![toolbar-1200](https://cloud.githubusercontent.com/assets/73419/8600485/774d92b2-2666-11e5-881e-9bc16c42008c.png)

### Ajax requests

![toolbar-ajax](https://cloud.githubusercontent.com/assets/73419/8600504/8ed69708-2666-11e5-8135-64e6e6c2ec52.gif)

### Toolbar Panels

#### Request

![panel-request](https://cloud.githubusercontent.com/assets/73419/8600515/9bd9c7fe-2666-11e5-914f-7f0f7e6a5979.png)

#### Time

![panel-time](https://cloud.githubusercontent.com/assets/73419/8600524/9f261714-2666-11e5-9dc8-53f9cae0f502.png)

#### Memory

![panel-memory](https://cloud.githubusercontent.com/assets/73419/8600529/a4bc3168-2666-11e5-9227-12ac8f44510c.png)

#### Twig

![panel-twig](https://cloud.githubusercontent.com/assets/73419/8600538/a8d4155e-2666-11e5-99b0-c9b5e8c82222.png)

#### Log

![panel-log](https://cloud.githubusercontent.com/assets/73419/8600542/ac2db854-2666-11e5-9e46-2c4c84e8521c.png)

#### Form

![panel-form](https://cloud.githubusercontent.com/assets/73419/8600549/afb75cc8-2666-11e5-9e61-0d1e99cc2201.png)

#### Translation

![panel-i18n](https://cloud.githubusercontent.com/assets/73419/8600558/b6e0d3da-2666-11e5-9e6d-c2178dff6204.png)

#### Security

![panel-security](https://cloud.githubusercontent.com/assets/73419/8600562/ba9a13d8-2666-11e5-9261-07670bcf98ca.png)

#### Database

![panel-db](https://cloud.githubusercontent.com/assets/73419/8600567/bf521f10-2666-11e5-9c94-293d97aa4522.png)

#### Dumper

![panel-dump](https://cloud.githubusercontent.com/assets/73419/8600570/c27e1234-2666-11e5-9d98-e680484edfbc.png)

#### Config

![panel-config](https://cloud.githubusercontent.com/assets/73419/8600573/c5c34e8c-2666-11e5-9cc4-9c99f4b26825.png)

javiereguiluz committed:
Fixed another insignificant syntax issue

mickaelandrieu commented:
@javiereguiluz this is realy nice, thank you for this amazing work ! Do you think your pull request may have impact on the extraction of Profiler ? I have the objective to "port" this profiler inside BackBee CMS with only small data to override and panels to add :+1:

ogizanagi commented:
@javiereguiluz : It looks awesome :+1:

skigun commented:
@javiereguiluz great job :+1: :clap:

xabbuh commented:
@javiereguiluz Overall I like your changes very much! :+1: I just think that we need to improve constrast a bit. I assume that especially the grey font on black background and the blue links will probably not be very readable for users with handicaps.

ogizanagi commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig:
```diff
@@ -1,41 +1,52 @@
+            </div>
+            <div class="sf-toolbar-info-piece">
+                <b>Controller file</b>
+                <span>{{ collector.controller.class }}</span>
```
Still an issue here.
```
{% if collector.controller.class is defined %}
```

javiereguiluz replied:
Fixed. Thanks!

ogizanagi commented:
I also have an issue with the `.sf-toolbar-block-config > .sf-toolbar-info` not being aligned right on profiler page :/

<img width="382" alt="screenshot 2015-07-09 a 20 28 16" src="https://cloud.githubusercontent.com/assets/2211145/8603514/1629d8a2-2679-11e5-84ee-cc21900e6c3a.PNG">

javiereguiluz committed:
Fixed a potential issue in the request panel

javiereguiluz committed:
A very high z-index value is required to avoid issues in the profiler view

javiereguiluz commented:
@ogizanagi the error in the profiler already happens in any Symfony version. It's not easy to solve. The magic that prevents this issue is included in `toolbar_js.html.twig`, which is not included in the profiler layout. I've tried to include it and causes side-effects and problems :(

ogizanagi commented:
@javiereguiluz : Why not simply adding a css rule such as:
```css
    @media (min-width: 768px) {
    .sf-toolbar-block-config:hover .sf-toolbar-info {
        right: 0;
    }
}
```
?

Also, another minor issue:
<img width="364" alt="screenshot 2015-07-09 a 23 10 36" src="https://cloud.githubusercontent.com/assets/2211145/8606984/012f3a48-2690-11e5-8059-71c49bfd224d.PNG">

shoomyth commented:
Awesome. But I think symfony version is not so important on each request as profile token for example

jorgelbg commented:
:+1: this looks awesome!

javiereguiluz committed:
Fixed another z-index issue

javiereguiluz committed:
Fixed an issue with the Config panel in the Profiler view

javiereguiluz committed:
Minor CSS tweaks and made font sizes bigger

javiereguiluz committed:
More vertical aligning fixes

javiereguiluz commented:
Update:

  * All `z-index` issues are fixed
  * When displayed inside the profiler, the toolbar panels now display correctly (thanks @ogizanagi for the tip)
  * Most vertical aligning issues are fixed
  * The font size in the information panels have been increased slightly to improve readability (thanks @xabbuh for the heads-up)

@shoomyth I'm curious about the profiler token. How do you use it? Why do you want to see it displayed in the toolbar? Thanks.

Pierstoval commented:
@javiereguiluz The profiler token can be used with XDebug and PHPStorm, for example, to profile each page load directly in the IDE

shoomyth commented:
@javiereguiluz hmm, iirc i need it to find token in wdt history.
I mean this symfony section is most about "static" parameters (server environment), but token, kernel name, app environment belong to each request and maybe should be placed on left side of toolbar? But at the same time I'm not sure they deserve own separate sections on it.

OskarStark commented:
@javiereguiluz amazing!

javiereguiluz committed:
fixed all vertical aligning issues and tweaked icons

javiereguiluz committed:
Made the close icon a bit smaller

javiereguiluz commented:
Is there anything left we can do before merging this change?

The reported Travis error isn't related to this PR:

```
1) Symfony\Component\Process\Tests\SigchildDisabledProcessTest::
   testIdleTimeoutNotExceededWhenOutputIsSent

A general timeout is expected.
Failed asserting that false is true.
/home/travis/build/symfony/symfony/src/Symfony/Component/Process/
Tests/AbstractProcessTest.php:855
```

hason commented:
Minor issue with padding:
![image](https://cloud.githubusercontent.com/assets/288535/8771557/71672a6a-2ec3-11e5-87ef-28a025d94a1c.png)

hason commented:
Issue in profiler:
![image](https://cloud.githubusercontent.com/assets/288535/8771634/434372a0-2ec4-11e5-8b37-9d2184bbe75b.png)

javiereguiluz committed:
Smaller font sizes for smartphones, fixed request status padding issue
and make too long panels always be displayed at the leftmost part of browser window

javiereguiluz committed:
Fixed toolbar issues when displaying it inside the profiler

javiereguiluz committed:
Minor tweaks

javiereguiluz commented:
@hason thanks for reporting these errors. They should be fixed now.

stof commented:
@javiereguiluz what would happen in this new design in case third-party panels still make things too wide for the screen ?

guilhermeblanco commented:
@javiereguiluz I would ask @hason to double check this again, but it seems Doctrine panel is showed twice in Profiler (as I can see in the provided screenshot). Is this issue still happening?

javiereguiluz commented:
@stof regarding the "too many panels issue", the new toolbar behaves exactly as the old one: if a panel doesn't fit, it's not displayed. Example:

Old toolbar at 1900px shows the SonataAdmin panel:

![old_1900](https://cloud.githubusercontent.com/assets/73419/8776075/a676e602-2eee-11e5-9675-98242b4d2fdb.png)

Old toolbar at 1200px doesn't show it:

![old_1200](https://cloud.githubusercontent.com/assets/73419/8776074/a173ccd8-2eee-11e5-8290-aa049827658c.png)

@guilhermeblanco I can't reproduce the issue. Maybe @hason is toying with the toolbar or using a panel which reuses the Doctrine icon.

javiereguiluz committed:
Pass the toolbar version number from the controller, to ease transition and keep BC

nesl247 commented:
Why not solve the "too many panels issue" like Zend's Z-Ray does? They use left/right arrows to be able to scroll.

hason commented:
@guilhermeblanco The first icon is for DoctrineBundle and second for PHPCRDoctrineBundle.

javiereguiluz committed:
Added some styles to make old panels look better in the new design

stof commented on src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php:
```diff
@@ -197,12 +201,15 @@ public function toolbarAction(Request $request, $token)
             // the profiler is not enabled
         }
 
+        $toolbarVersion = Kernel::VERSION_ID < 20800 ? self::TOOLBAR_VERSION_1 : self::TOOLBAR_VERSION_2;
```
this does not make sense. You are choosing the toolbar version based on the HttpKernel component being installed, while it actually depends on the WebProfilerBundle version (templates and styles are in WebProfilerBundle). So when this code is used, it will always be version 2 (until the day we implement a third iteration)

OskarStark commented on src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php:
```diff
@@ -134,4 +136,49 @@ protected function templateExists($template)
+        $templates = $this->moveArrayElementToFirstPosition($templates, 'memory');
+        $templates = $this->moveArrayElementToFirstPosition($templates, 'time');
+        $templates = $this->moveArrayElementToFirstPosition($templates, 'request');
+
```
why do you have a blank line here?

Soullivaneuh commented:
@hason About the profiler, I think the design should be also review to be like the new toolbar, isn't it?

razmo commented:
@javiereguiluz nice job :+1:

stof commented on src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php:
```diff
@@ -134,4 +136,49 @@ protected function templateExists($template)
+     *
+     * @return array
+     */
+    protected function reorderTemplates($templates)
```
please make it private

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig:
```diff
@@ -1,41 +1,54 @@
             <div class="sf-toolbar-info-piece">
-                <b>Controller</b>
-                {{ request_handler }}
+                <b>Controller method</b>
```
this change is wrong IMO. It is not always a method. It is any callable

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig:
```diff
@@ -1,41 +1,54 @@
+            {% if collector.controller.class is defined %}
+                <div class="sf-toolbar-info-piece">
+                    <b>Controller file</b>
+                    <span>{{ collector.controller.class }}</span>
```
this does not make sense. a file and a class are not the same

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/time.html.twig:
```diff
@@ -16,18 +16,28 @@
-    {% set duration = collector.events|length ? '%.0f ms'|format(collector.duration) : 'n/a' %}
+    {% set total_time = collector.events|length ? '%.0f'|format(collector.duration) : 'n/a' %}
+    {% set initialization_time = collector.events|length ? '%.0f'|format(collector.inittime) : 'n/a' %}
+    {% set status_color = collector.events|length ? collector.duration > 1000 ? 'yellow' : '' %}
```
I suggest using ``and`` rather than nested ternary operators

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -165,31 +165,26 @@
-                        row.appendChild(cell);
-                        tbody.appendChild(row);
+                        var ajaxToolbarPanel = document.querySelectorAll('.sf-toolbar-block-ajax');
+                        ajaxToolbarPanel[0].style.display = 'none';
```
just use ``querySelector`` rather than ``querySelectorAll()[0]``

stof replied:
and shouldn't you reset the display later to show it again ?

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -165,31 +165,26 @@
+                var className = 'sf-toolbar-ajax-requests sf-toolbar-value';
+                requestCounter[0].className = className;
+
+                var ajaxToolbarPanel = document.querySelectorAll('.sf-toolbar-block-ajax');
```
you should avoid querying twice for this block. and you should move the DOM traversal before the DOM changes done above as done elsewhere to invalidate the DOM only once

javiereguiluz committed:
Misc. tweaks and improvements

javiereguiluz committed:
Minor JavaScript optimizations

Mondane commented:
Does this redesign show the file and line number again in the profiler? I'm missing that in 2.7 but as I read this, it was there in 2.2: http://symfony.com/blog/new-in-symfony-2-2-logging-of-deprecated-calls . Now I have to open the log file to see the file for the message.

javiereguiluz committed:
Reverted the feature to display different toolbar versions

javiereguiluz committed:
Removed an unused import

javiereguiluz commented:
All reported errors seem to be fixed now. What about merging this PR as it is so more people can easily test the new toolbar and report any other error?

stof commented:
@Mondane the profile still displays these informations in 2.7 when they are available. but the error handler does not pass the stack trace to the logger anymore, so there is no info to display. It is not a profiler issue.

@javiereguiluz what would be the way for a bundle to know which version of the toolbar markup it should provide ? You don't seem to provide a way to detect that WebProfilerBundle 2.8+ is used (and no, using the kernel version in the template is not a working solution as bundles and components can be installed at different versions)

stof commented:
and what are you plans for the design of the profiler itself. Will it be done in a separate PR or not at all ?

fabpot commented:
I think that the profiler design is another topic that could be done in another PR.

javiereguiluz commented:
@stof I appreciate your patience and your code review in the PR that I submitted to DoctrineBundle. However, the sad truth is that I don't understand you :(  If someone who understands this problem well can send me the needed code, I'll integrate it right away. Thanks!

WouterJ commented:
@javiereguiluz it is possible to use HttpKernel 2.3 with WebProfilerBundle 2.8 (for instance, in a Silex app). So relying on the HttpKernel version to choose the web profiler bundle isn't possible.

stof commented:
@javiereguiluz you need to pass a variable to the template which would allow the bundle to know that the new WebProfilerBundle is calling the template. And the value of the variable should be something future-proof (i.e. being able to change the value for a future redesign). Bundles would then assume that when this variable is not available, it is the old version of the bundle (as symfony 2.7- is not passing this variable).

Here is how it could work:

- Symfony 2.7 and older: no ``webprofiler_markup`` variable
- Symfony 2.8: ``webprofiler_markup = 2``
- future Symfony version redesigning the profiler again: ``webprofiler_markup = 3``
- even future version of Symfony: ``webprofiler_markup = 4``

And then DoctrineBundle could support all versions like this (in my example, it supports version 1, 2 and 3 but not yet version 4):

```jinja
{# ensure the variable is defined even for Symfony 2.7- #}
{% set webprofiler_markup = webprofiler_markup|default(1) %}

{% if webprofiler_markup = 1 %}
    {# markup for the WebProfilerBundle 2.7- (first design of the profiler)
{% elseif webprofiler_markup = 2 %}
    {# markup for the WebProfilerBundle 2.8+ (second design of the profiler)
{% else %}
    {# markup for newer version, coded against webprofiler_markup=3 but
        displayed for any higher version so that a new webprofiler version would
       get the markup of the previous one instead of nothing, until the template is
       updated to support the new version 4.
     #}
{% endif %}
```

note the order of the different markup version in the template, to ensure that when we introduce a new design of the profiler, bundles not updated to support them would provide the previous markup because it is better to have a slightly broken design rather than displaying nothing.

The ``webprofiler_markup`` variable should be bumped when doing design changes in the WebProfilerBundle which require to change the markup of toolbar items, so it should not happen too often.

javiereguiluz committed:
Added a new profiler_markup_version to improve BC of the new toolbar

javiereguiluz commented:
@stof thanks for this new explanation. I've made the required changes in 5070861 and https://github.com/doctrine/DoctrineBundle/pull/445   Should I also do the same changes for SwiftmailerBundle and SecurityBundle?

stof commented:
@SecurityBundle does not need it IMO as it is in the symfony repo. SwiftmailerBundle would need an update of the template, yes

WouterJ commented:
Is it an idea to add a summary of the markup changes in the UPGRADE guide?

javiereguiluz committed:
Removed an useless CSS class and added styles for <hr>

stof commented:
it would be great yeah

fabpot commented:
:+1: Let's merge this one

stof commented:
I would like to have the description of markup changes in the upgrade file. but otherwise, :+1:

stof commented on src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig:
```diff
@@ -2,39 +2,47 @@
+    {% set icon %}
+        <svg version="1.1" xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" height="24" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
+            <path fill="#AAAAAA" d="M21,20.4V22H3v-1.6c0-3.7,2.4-6.9,5.8-8c-1.7-1.1-2.9-3-2.9-5.2c0-3.4,2.7-6.1,6.1-6.1s6.1,2.7,6.1,6.1c0,2.2-1.2,4.1-2.9,5.2C18.6,13.5,21,16.7,21,20.4z"/>
+        </svg>
```
WebProfilerBundle is putting SVG icons in separate templates. What about doing the same in SecurityBundle and DebugBundle to be consistent ?

javiereguiluz replied:
I'd prefer to not do that. In the other icons make sense because we have a folder with lots of icons. In this case it woul be only 1 icon in 1 folder. I don't like that.

stof replied:
There would be an advantage if we rename icon files to be named ``.svg`` as I suggested in my other comment.

and icons are also used in the left menu of the profiler. In older versions of Symfony, we are duplicating the inline icon in both places. Having it in an include file allows to reuse the icon without duplication.

javiereguiluz commented:
@stof give me 1 minute and I'll add the changes in UPGRADE file.

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig:
```diff
@@ -1,115 +1,103 @@
 
-    {# PHP Information #}
-    {% set icon %}
-        <a href="{{ path('_profiler_phpinfo') }}">
```
do we still have a way to access the phpinfo page anywhere ?

javiereguiluz replied:
Good catch! I've readded it as a link next to the PHP version:

![phpinfo](https://cloud.githubusercontent.com/assets/73419/9005704/c242da68-3780-11e5-8429-fde31afe87e7.png)

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/form.html.twig:
```diff
@@ -3,13 +3,31 @@
+            </div>
+            <div class="sf-toolbar-info-piece">
+                <b>Number of errors</b>
+                <span class="sf-toolbar-status sf-toolbar-status-{{ collector.data.nb_errors > 0 ? 'red' }}">{{ collector.data.nb_errors }}</span>
```
you should avoid putting a ``sf-toolbar-status-`` class when there is not error. The whole class should go in the conditional instead of putting only the suffix in it, given that you don't put any color anymore on valid forms

javiereguiluz replied:
I prefer to not make this change. It's not a bug and "fixing" it will complicate Twig templates.

stof replied:
complicate them ?

```html+twig
<span class="sf-toolbar-status sf-toolbar-status-{{ collector.data.nb_errors > 0 ? 'red' }}">
{# vs #}
<span class="sf-toolbar-status {{ collector.data.nb_errors > 0 ? 'sf-toolbar-status-red' }}">
```

javiereguiluz replied:
We usually have two different values in the ternary condition, so the templates would become more verbose:

```twig
<span class="sf-toolbar-status sf-toolbar-status-{{ collector.data.nb_errors > 0 ? 'red' : 'green' }}">
{# vs #}
<span class="sf-toolbar-status {{ collector.data.nb_errors > 0 ? 'sf-toolbar-status-red' : 'sf-toolbar-status-green' }}">
```

WouterJ replied:
@javiereguiluz for panels, you removed all "green" statuses, so it doesn't apply to much cases anymore.

Btw, if  this is so common, why not creating a symfony filter/function to handle this?

stof replied:
@javiereguiluz my comment was related to the case where the else clause displays nothing. When both cases are showing a color, the current way is perfectly fine (which is why it was used before btw)

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/logger.html.twig:
```diff
@@ -86,7 +80,7 @@
     </table>
 
-    {% if collector.logs %}
+{% if collector.logs %}
```
this change should be reverted

javiereguiluz committed:
Added some upgrade notes about the new toolbar design

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig:
```diff
@@ -5,36 +5,34 @@
+            <div class="sf-toolbar-info-piece">
+                <b>Missing messages</b>
+                <span class="sf-toolbar-status sf-toolbar-status-{{ collector.countMissings ? 'red' }}">
+                    {{ collector.countMissings|default(0) }}
```
useless default filter. It is always defined and always return an integer

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig:
```diff
@@ -5,36 +5,34 @@
+            <div class="sf-toolbar-info-piece">
+                <b>Fallback messages</b>
+                <span class="sf-toolbar-status sf-toolbar-status-{{ collector.countFallbacks ? 'yellow' }}">
+                    {{ collector.countFallbacks|default(0) }}
```
useless default filter. It is always defined and always return an integer

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig:
```diff
@@ -5,36 +5,34 @@
+
+            <div class="sf-toolbar-info-piece">
+                <b>Defined messages</b>
+                <span class="sf-toolbar-status">{{ collector.countdefines|default(0) }}</span>
```
useless default filter. It is always defined and always return an integer

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig:
```diff
@@ -5,36 +5,34 @@
+            </div>
+
+            <div class="sf-toolbar-info-piece">
+                <b>Defined messages</b>
```
the text can be confusing here. These are the defined messages among the translations being used. you can have much more defined messages in your translation file

javiereguiluz replied:
The title can be improved, but it's the same title that Symfony has used since the beginning and the profiler uses too. I guess it's better not to change it in this PR.

javiereguiluz replied:
By the way, I'm getting this error when I click on the "i18n" icon to open the "i18n" profiler:

![i18n_profiler](https://cloud.githubusercontent.com/assets/73419/9006408/3d67e08e-3787-11e5-83af-b9788fa6f748.png)

This error occurs in some code which hasn't been modified by this PR. What should we do? Maybe @aitboudad can help us spotting the problem.

stof replied:
@javiereguiluz I see 2 possible causes here:

- you are using some data profiled on Symfony 2.7 (or with 2.8 before the merge of #15139) and viewing it on Symfony 2.8. This is not supported. We don't preserve BC on profile data to ensure that templates can display old data where we could have less things (we preserve BC on available data so that code reading data will still work). The template requires the 2.8+ data
- you are using the Translator component at version 2.7, and so collected data look like the 2.7 ones

javiereguiluz replied:
@stof thanks. It was one of the problems that you explained. Everything is right now :)

stof replied:
> The title can be improved, but it's the same title that Symfony has used since the beginning

this actually means "that Symfony is using in 2.7" as this panel is new in 2.7

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Icon/twig.html.twig:
```diff
@@ -0,0 +1,5 @@
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" height="24" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
```
the file should be named with ``.svg``, not with ``.html``.

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -165,31 +165,26 @@
+                var className = 'sf-toolbar-ajax-requests sf-toolbar-value';
+                requestCounter[0].className = className;
+
+                var ajaxToolbarPanel = document.querySelector('.sf-toolbar-block-ajax');
```
you should query for the toolbar panel only once. Here, you do it on line 172 and here.

and the DOM query should be done at the beginning of the method, with other DOM queries, to avoid invalidating the DOM several times

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -26,127 +17,140 @@
+    color: #AAA;
+    font-size: 12px;
+}
+/* TODO: remove it when Twig uses SVG icons */
```
doesn't it use one ?

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Icon/form.svg.twig:
```diff
@@ -0,0 +1,6 @@
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" height="24" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
```
these files are pure SVG files, with no twig markup in them. I suggest removing the ``.twig`` extension on them. This way, tools will consider them as SVG files (github will probably show a preview for instance).
Twig is able to include any file anyway (the requirement for a ``.twig`` extension comes from the Templating component to support multiple engines, not from Twig, and we don't use the Templating component for these templates)

stof commented:
I actually made a couple more comments

stof commented:
@javiereguiluz Icons are used in 2 places in the WebProfiler: in the toolbar items, and in the left menu of the profiler. As you are changing icons in the toolbar here, we now have inconsistent icons in the profiler. Should we keep it like that until the profiler itself is redesigned, or should we use the new icons in the left menu already to have consistency ?

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -323,121 +307,168 @@
+    50% { background: #444; }
+    100% { background: #222; }
 }
 @-o-keyframes sf-blink {
```
``-o-keyframes`` is useless. It was never supported prefixed before Opera switched their rendering engine

javiereguiluz committed:
Tweaks and bug fixes

stof commented:
> egarding the "too many panels issue", the new toolbar behaves exactly as the old one: if a panel doesn't fit, it's not displayed

@javiereguiluz this is not the behavior of the old one. On Symfony 2.6.4+, the toolbar spans multiple lines instead of having items being placed off screen: https://github.com/symfony/symfony/pull/12649

stof commented on src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php:
```diff
@@ -134,4 +136,49 @@ protected function templateExists($template)
+     *
+     * @return array
+     */
+    private function reorderTemplates($templates)
```
wouldn't it be better to change the priority when registering templates to register them in the right order directly ? this would give more control over the order

javiereguiluz replied:
Yes. I don't like this solution either. Please tell me where can I configure this priority. Thanks.

javiereguiluz replied:
I've just made this change ... which has allowed us to remove a lot of lines of code. \o/  Thanks!

javiereguiluz committed:
Restored the old behavior for toolbars with lots of elements

If a toolbar includes lots of elements, some of them may
span to a new line, increasing the height of the toolbar

WouterJ commented:
There are some BC breaks in the menu that are not in the update file:

 * `<div class="sf-toolbar-icon">` now has to live inside the `<a>` (which is incorrect btw, but is supported by browsers), whereas the other way around was supported.
 * Toolbars now have to have an icon (or at least an `<img>` element), otherwise `.sf-toolbar-value/label` aren't styled correctly.

javiereguiluz committed:
Redesigned "abbr" elements

stof commented:
@WouterJ a div inside a a is not invalid anymore in the HTML5 spec (it was indeed invalid in the HTML4 spec, while supported by browsers)

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -26,127 +17,135 @@
+.sf-toolbar-block .sf-toolbar-info {
+    border-collapse: collapse;
+    display: table;
+    z-index: 99999;
```
the toolbar info should have a higher zindex than the toolbar itself to place it properly when the toolbar spans on 2 lines (ensuring that the info does not get partially hidden by the first row of the toolbar)

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -323,121 +302,158 @@
+        right: 0;
+    }
+    .sf-toolbar-block-time .sf-toolbar-icon svg,
+    .sf-toolbar-block-memory .sf-toolbar-icon svg {
```
rather than hardcoding this handling for the time and memory panel, it would be better to use an agnostic class which could be opted in by any panel wanting a similar behavior (and this behavior would be used by the time and memory ones of course)

stof replied:
the more we can avoid panel-specific styles in the stylesheet, the more our toolbar will be extendable. and given it is meant to be fully extendable, it is a good goal.

javiereguiluz replied:
For now I prefer to not make this changes. The reason is that design of the time + memory panels is custom and totally different from the other panels. Why? Because I think that time+memory should be in fact one single panel called "performance". All thse CSS twekas are defined to make these panels look like one single panel.

stof replied:
yeah, and the memory section does not even have a panel in the profiler right ? It is only in the toolbar

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig:
```diff
@@ -323,121 +302,158 @@
     }
 }
+
+@media (min-width: 1200px) {
```
if there is nothing to put in it, I suggest omitting it entirely

javiereguiluz committed:
Removed an unused media query

javiereguiluz committed:
Increased the z-index of .sf-toolbar-info

javiereguiluz commented:
@WouterJ I can't reproduce this:

> Toolbars now have to have an icon (or at least an <img> element), otherwise .sf-toolbar-value/label aren't styled correctly.

Panels without icons (svg or img) look OK.

WouterJ commented:
Hmm, indeed. Must have been something else than that broke SonataBlockBundle's toolbar block

javiereguiluz committed:
Reordered the toolbar elements via service priorities

javiereguiluz commented:
Update:

  * All reported errors have been fixed.
  * All proposed changes have been implemented.

The error displayed by Travis for PHP 5.3 is unrelated to this PR:

1) Symfony\Component\Console\Tests\Helper\ProgressBarTest::testAnsiColorsAndEmojis
Failed asserting that two strings are equal.

WouterJ commented:
:+1: Imo, it's better to merge this PR now. We have 4 months to improve the current toolbar in further PRs.

fabpot commented:
Thank you @javiereguiluz.

stof commented on src/Symfony/Bundle/FrameworkBundle/Resources/config/collectors.xml:
```diff
@@ -17,17 +17,17 @@
         <service id="data_collector.request" class="%data_collector.request.class%">
             <tag name="kernel.event_subscriber" />
-            <tag name="data_collector" template="@WebProfiler/Collector/request.html.twig" id="request" priority="255" />
+            <tag name="data_collector" template="@WebProfiler/Collector/request.html.twig" id="request" priority="260" />
```
@javiereguiluz I would even suggest leaving holes between the priorities used in Symfony to make it easier to inject things between them

peshi commented:
@javiereguiluz The close toolbar "X" is misaligned in OS X Safari.

javiereguiluz commented:
@peshi thanks! I've created a meta-issue to collect all the issues/bugs related to the redesign: https://github.com/symfony/symfony/issues/15439

tacman commented:
This is great.  My Propel SQL statements are now with the HTML tags though, is there a |raw that is missing somewhere?  Or a setting I can use?

![image](https://cloud.githubusercontent.com/assets/619585/12068747/6e102e20-afe2-11e5-8753-79f98f18ec55.png)

sstok commented:
@tacman This should be reported to the https://github.com/propelorm/PropelBundle (if not done already).

hrazaka commented:
I have some problem.
I have double click on sf toolbar and it becomes full screen.
I am not able to view my page. I always see sf toolbar full screen.
Please help to reduce the sf toolbar size
